packages/react-native-bridge/android/react-native-bridge/src/main/java/org/wordpress/mobile/ReactNativeGutenbergBridge/RNReactNativeGutenbergBridgeModule.java

```diff
@@ -554,14 +556,59 @@ private ConnectionStatusCallback requestConnectionStatusCallback(final Callback
         };
     }
 
+    @ReactMethod
+    public void showAndroidSoftKeyboard(Integer delayMillis) {
+        Activity currentActivity = mReactContext.getCurrentActivity();
+        if (currentActivity != null) {
+            View currentFocusedView = currentActivity.getCurrentFocus();
+            if (currentFocusedView != null) {
+                // Cancel any previously scheduled Runnable
+                if (mKeyboardRunnable != null) {
+                    currentActivity.getWindow().getDecorView().removeCallbacks(mKeyboardRunnable);
+                }
+
+                final Runnable showTheKeyboardNow = new Runnable() {
+                    @Override
+                    public void run() {
+                        try {
+                            View activeFocusedView = currentActivity.getCurrentFocus();
+                            if (activeFocusedView != null && currentActivity.getWindow().getDecorView().isShown()) {
+                                InputMethodManager imm =
+                                    (InputMethodManager) mReactContext.getSystemService(Context.INPUT_METHOD_SERVICE);
+                                imm.showSoftInput(activeFocusedView, InputMethodManager.SHOW_IMPLICIT);
+                            }
+                        } catch (Exception e) {
+                            // Noop
+                        }
+                    }
+                };
+
+                // Schedule the keyboard showing with a delay
+                if (currentFocusedView.hasWindowFocus()) {
+                    currentActivity.getWindow().getDecorView().postDelayed(showTheKeyboardNow, delayMillis);
+                } else {
+                    currentFocusedView.getViewTreeObserver().addOnWindowFocusChangeListener(new ViewTreeObserver.OnWindowFocusChangeListener() {
+                        @Override
+                        public void onWindowFocusChanged(boolean hasFocus) {
+                            if (hasFocus) {
+                                currentActivity.getWindow().getDecorView().postDelayed(showTheKeyboardNow, delayMillis);
+                                currentFocusedView.getViewTreeObserver().removeOnWindowFocusChangeListener(this);
+                            }
+                        }
+                    });
+                }
+            }
+        }
+    }
```

Comment:
Nitpick: After checking [this code](https://github.com/wordpress-mobile/WordPress-Android/blob/6e653b566c96970ebbad1f88e4574ac6a5c518fb/WordPress/src/main/java/org/wordpress/android/util/extensions/ViewExtensions.kt#L46-L79), I wonder if we could extract to separate functions the logic of to simplify this function:
- `isAnyViewFocused`: Checks if there's any view focused for the current activity.
- `getCurrentFocusedView`: Returns the current focused view.

Reply:
Good idea! I'll also extract the runnable to a separate function

Reply:
> When I was working on this I realized `mKeyboardRunnable` wasn't being set 😅

Oh, I overlooked this too. Thanks for addressing it 🙇 !

> I made a few changes in https://github.com/WordPress/gutenberg/commit/b0bfdf10f48440d69d585aeb379c5c9ad9fe4f56 where it splits the large function into smaller functions, with this refactor and setting the missing mKeyboardRunnable I realized the delay wasn't needed for the full-screen modal 🚀

That's great 🎊. Was the reason for needing the delay the fact of not setting `mKeyboardRunnable` variable?

Reply:
> That's great 🎊. Was the reason for needing the delay the fact of not setting mKeyboardRunnable variable?

Not necessarily it was an outcome of refactoring the whole thing 😄